Removing Runtime Babel Dependency

PierreAndreis opened the pull request:
I have update babel presets to the latest, removed all runtime babel dependency and republished it

I have also raised the requirements to node 6.

Reason is because most of the ES5/6 stuff are already supported by Node 6, there is no need for run time babel. It just makes the application slower with the addition of numerous boilerplates

seripap committed:
v1.3.7 - Support for Vainglory v2.12

seripap committed:
1.3.8

seripap committed:
v1.3.8

seripap committed:
Update CHANGELOG.md

PierreAndreis committed:
Adding ranked 5v5 support

cklugewicz committed:
Corrected display name for item TotemEnhancer (#31)

seripap committed:
1.4.0

PierreAndreis committed:
Removing dependency on babel and jest

PierreAndreis committed:
Running pre-publish

PierreAndreis committed:
Removing babel run-time

PierreAndreis committed:
Merge branch 'master' into master

seripap commented:
Thanks, I'll have time this weekend to take a look and get back to you

PierreAndreis commented:
@seripap If this is a major change, I could also get this opportunity to work on others requested features such as #28 #10

seripap commented:
@PierreAndreis sure thing, feel free to tackle whatever you can assist with 👍

seripap commented:
Any updates on your side for more changes? I may actually re-work portions so that babel is removed entirely. A lot of things are supported natively these days and it doesn't make sense to babelify a server side library

PierreAndreis commented:
@seripap I haven't made any change so far, but I agree with you on removing babel completely. 
Only babel exclusive that node doesn't support yet is import/export (which babel is compiling to require... so no difference)

PierreAndreis commented:
I can work on some changes tonight. 
Basically I want to get a .JSON() method for each class so we can get a json including getter/setter
which would fix https://github.com/seripap/vainglory/issues/10

nathancarter committed:
Adding documentation to undocumented attributes (#36)

* Adding documentation to undocumented attributes

Hopefully these docs can help others who browse to this page and want to know what the various stats are and what they mean.  Not all meanings were clear to me from data I've inspected, so I'm leaving some unspecified.  Obviously anyone can come along later and propose a change to add info I didn't know how to add.

* Improving this PR based on help from PierreAndreis
